src/Common/src/Interop/Windows/mincore/Interop.VersionHelpers.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Kernel32)]
+        private static extern bool VerifyVersionInfo([In] ref OSVERSIONINFOEX lpVersionInfo, uint dwTypeMask, ulong dwlConditionMask);
+
+        [DllImport(Libraries.Kernel32)]
+        private static extern ulong VerSetConditionMask(ulong dwlConditionMask, uint dwTypeBitMask, byte dwConditionMask);
```

Comment:
The Win32 APIs will lie.  On Windows 10, they won't return a 10.0 version number.  You need to add a Windows manifest to your application and put the proper GUIDs in it.  Then Windows won't lie to you anymore